prometheus-to-sd/translator/translator_test.go

```diff
@@ -416,6 +426,136 @@ func TestTranslatePrometheusToStackdriver(t *testing.T) {
 	}
 }
 
+func TestTranslateSummary(t *testing.T) {
+	var intSummaryMetricsResponse = &PrometheusResponse{rawResponse: `
+# TYPE process_start_time_seconds gauge
+process_start_time_seconds 1234567890
+# TYPE int_summary_metric summary
+int_summary_metric{quantile="0.5"} 4
+int_summary_metric{quantile="0.9"} 8
+int_summary_metric{quantile="0.99"} 8
+int_summary_metric_sum 42
+int_summary_metric_count 101010
+`}
+	var floatSummaryMetricsResponse = &PrometheusResponse{rawResponse: `
+# TYPE process_start_time_seconds gauge
+process_start_time_seconds 1234567890
+# TYPE float_summary_metric summary
+float_summary_metric{quantile="0.5"} 4.12
+float_summary_metric{quantile="0.9"} 8.123
+float_summary_metric{quantile="0.99"} 8.123
+float_summary_metric_sum 0.42
+float_summary_metric_count 50
+`}
+
+	type expectedMetricAttributes struct {
+		name         string
+		resourceType string
+		valueType    string
+		doubleValue  float64
+		intValue     int64
+		startTime    string
+		endTime      string
+	}
+	type summaryTest struct {
+		description        string
+		prometheusResponse *PrometheusResponse
+		summaryMetricName  string
+		expectedAttributes []expectedMetricAttributes
+	}
+
+	sts := []summaryTest{
+		{
+			description:        "Test summary metrics which accumulate integer values",
+			prometheusResponse: intSummaryMetricsResponse,
+			summaryMetricName:  intSummaryMetricName,
+			expectedAttributes: []expectedMetricAttributes{
```

Comment:
You can replace expectedAttributes by list of timeseries, and then all your logic below is going to be simplified to comparision of two objects.

Reply:
ok.

Reply:
Added.